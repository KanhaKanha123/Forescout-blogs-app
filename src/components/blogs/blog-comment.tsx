import { FC, ReactElement } from "react";
import { BlogCommentWrapper, Text, UserNameWrapper, Image, BlogCommentTextWrapper } from "./style";
import userImg from "../../assets/icons/user-Icon.png";
import { BlogSingleCommentTypes } from "../../custom-hooks/use-get-blog-comments";

export interface SingleBlogCommentTypes {
    readonly comment: BlogSingleCommentTypes
};

/**
 * This component is responsible to render the blog single comment and user details who added the comment.
 * @params comment
 * @returns ReactElement
 */
export const BlogComment: FC<SingleBlogCommentTypes> = ({ comment }): ReactElement => {

    return <BlogCommentWrapper aria-label="Single comment container">
        <UserNameWrapper aria-label="User name and user icon contaainer">
            <Image aria-label="user icon" src={userImg}></Image>
            <Text aria-label="user name">{comment.user.username}</Text>
        </UserNameWrapper>
        <BlogCommentTextWrapper aria-label="comment detail description">{comment.body}</BlogCommentTextWrapper>
    </BlogCommentWrapper>
};