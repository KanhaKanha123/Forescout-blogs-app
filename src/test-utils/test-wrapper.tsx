import ContextProvider from '../store/context';
import { QueryClientProvider } from "react-query";
import { FC, ReactElement } from 'react';
import { client } from '../react-query/query-client';
import { BrowserRouter as Router } from 'react-router-dom';

interface TestWrapperProps {
    children: React.ReactNode
}

/**
 * This component is responsible to wrap the tests with all the important wrappers so all the time in all tests we don't have to repeat.
 * @returns ReactElement
 */
export const TestWrapper: FC<TestWrapperProps> = ({ children }): ReactElement => (
    <QueryClientProvider client={client}>
        <ContextProvider>
            <Router>
                {children}
            </Router>
        </ContextProvider>
    </QueryClientProvider>);