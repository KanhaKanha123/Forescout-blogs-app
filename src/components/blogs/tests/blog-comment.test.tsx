import { render } from '@testing-library/react';
import { BlogComment } from '../blog-comment';
import { TestWrapper } from '../../../test-utils/test-wrapper';
import { commentData } from './fixture/mock-data';

const rederComponent = () => render(<TestWrapper>
  <BlogComment comment={commentData} />
</TestWrapper>);

describe('BlogComment component', () => {

  test('renders BlogComment component proper', () => {

    const component = rederComponent();

    expect(component).toMatchSnapshot();
  });

  test('renders BlogComment component should have proper props', () => {

    const component = rederComponent();

    const commentBody = component.getByText(/test comment/i);

    const testName = component.getByText(/test name/i);

    expect(commentBody).toBeInTheDocument();

    expect(testName).toBeInTheDocument();
  });
});
