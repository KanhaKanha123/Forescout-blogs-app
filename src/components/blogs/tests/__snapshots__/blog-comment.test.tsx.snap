// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogComment component renders BlogComment component proper 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="Single comment container"
        class="sc-jrICsz dOSEUl"
      >
        <div
          class="sc-kEbfvU eijDqn"
        >
          <img
            aria-label="user icon"
            class="sc-iCEALv cNPWGl"
            src="user-Icon.png"
          />
          <span
            aria-label="user name"
            class="sc-crPhKe cUNDRE"
          >
            test name
          </span>
        </div>
        <span
          aria-label="comment detail description"
          class="sc-iqkjcl bXffyC"
        >
          test comment
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="Single comment container"
      class="sc-jrICsz dOSEUl"
    >
      <div
        class="sc-kEbfvU eijDqn"
      >
        <img
          aria-label="user icon"
          class="sc-iCEALv cNPWGl"
          src="user-Icon.png"
        />
        <span
          aria-label="user name"
          class="sc-crPhKe cUNDRE"
        >
          test name
        </span>
      </div>
      <span
        aria-label="comment detail description"
        class="sc-iqkjcl bXffyC"
      >
        test comment
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
